scripts/check_compliance.py

```diff
@@ -588,7 +589,16 @@ def run(self):
 
                 original_fp = str(file['path']).replace('scancode-files/', '')
                 licenses = file['licenses']
-                if (file['is_script'] or file['is_source']) and (file['programming_language'] not in whitelist_languages) and (file['extension'] not in whitelist_extensions):
+                is_script = file.get("is_script")
+                is_source = file.get("is_source")
+                is_media = file.get("is_media")
+                pl = file.get("programming_language", None)
+                extension = file.get("extension")
+                file_type = file.get("file_type")
+                kconfig = "Kconfig" in original_fp and file_type in ['ASCII text']
+                valid_type = is_script or is_source or pl in blacklist_languages or not is_media or kconfig
```

Comment:
This will be true for any file with `not file.get("is_media")`. Is that deliberate?

Could maybe use this style instead of creating lots of variables that are only used in a single place:

```python
checked_file = (file.get('is_script') or
                file.get('is_source') or
                ...)
```

Makes it clear where the value is used, and that it's only used once.

Reply:
> This will be true for any file with `not file.get("is_media")`. Is that deliberate?

yes, I do not want to report anything related to media files.

Reply:
But you want to report everything for everything that's not a media file? That's what the logic is now.

If scripts aren't media files, then those would automatically get included by the `or not is_media`, for example.

Reply:
ah, you are right.